src/main/java/com/capgemini/wsb/fitnesstracker/training/internal/TrainingServiceImpl.java

```diff
@@ -1,16 +1,84 @@
 package com.capgemini.wsb.fitnesstracker.training.internal;
+import com.capgemini.wsb.fitnesstracker.training.api.TrainingNotFoundException;
+import com.capgemini.wsb.fitnesstracker.training.api.TrainingService;
+import com.capgemini.wsb.fitnesstracker.training.api.dto.NewTrainingDto;
+import com.capgemini.wsb.fitnesstracker.user.api.dto.*;
+import com.capgemini.wsb.fitnesstracker.training.api.Training;
+import com.capgemini.wsb.fitnesstracker.training.api.dto.TrainingDto;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.stereotype.Service;
 
-import com.capgemini.wsb.fitnesstracker.training.api.TrainingProvider;
-import com.capgemini.wsb.fitnesstracker.user.api.User;
-
+import java.util.Date;
+import java.util.List;
 import java.util.Optional;
 
 // TODO: Provide Impl
-public class TrainingServiceImpl implements TrainingProvider {
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class TrainingServiceImpl implements TrainingService {
+    private final TrainingMapper trainingMapper;
+    private final TrainingRepository trainingRepository;
+    private final BCryptPasswordEncoder bCryptPasswordEncoder;
 
     @Override
-    public Optional<User> getTraining(final Long trainingId) {
+    public Optional<Training> getTrainingById(final Long trainingId) {
         throw new UnsupportedOperationException("Not finished yet");
     }
 
-}
+    @Override
+    public List<TrainingDto> findAllTrainings() {
+        return trainingRepository.findAll()
+                .stream()
+                .map(trainingMapper::toTrainingDto)
+                .toList();
+    }
+
+    @Override
+    public List<Training> getTrainingsByUser(UserDto user) {
```

Comment:
Consider renaming method to something like `findTrainingsByUser`. In Java get is used mostly for getters.